### PR TITLE
[SPARK-18818][PYTHON] Add 'ascending' parameter to Window.orderBy()

### PR DESCRIPTION
**Pull request by ghost**
## What changes were proposed in this pull request?

Pass `ascending` parameter to Window's orderBy() function the same like DataFrame's orderBy() accepts. 
This allows to specify descending order without building Column object.
It makes this function more consistent and convenient.

## How was this patch tested?

Unit tests were run. Docstring tests were added to Window.orderBy()

### CODE REVIEW

File: python/pyspark/sql/window.py [hunk 1]
```diff
@@ -76,12 +79,37 @@ def partitionBy(*cols):
 
     @staticmethod
     @since(1.4)
-    def orderBy(*cols):
+    def orderBy(*cols, **kwargs):
         """
         Creates a :class:`WindowSpec` with the ordering defined.
+
+        :param cols: names of columns or expressions.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/window.py]: We should add ` ... versionchanged:: 2.5 blah blah` BTW

File: python/pyspark/sql/window.py [hunk 2]
```diff
@@ -76,12 +79,37 @@ def partitionBy(*cols):
...
+        |  2|       a|  2|
+        |  1|       a|  3|
+        |  1|       a|  4|
+        +---+--------+---+
         """
         sc = SparkContext._active_spark_context
-        jspec = sc._jvm.org.apache.spark.sql.expressions.Window.orderBy(_to_java_cols(cols))
+        ascending = kwargs.get('ascending', True)
+        jspec = sc._jvm.org.apache.spark.sql.expressions.Window.orderBy(
+            _to_seq(sc, _to_sorted_java_columns(cols, ascending)))
```
Comment (HyukjinKwon) [inline python/pyspark/sql/window.py]: Does Scala side have `orderBy` as well?

File: python/pyspark/sql/window.py (hunk 1 above)
Comment (annamolchanova) [reply python/pyspark/sql/window.py]: Hi @HyukjinKwon , I've added version directive. Thanks

File: python/pyspark/sql/window.py (hunk 2 above)
Comment (annamolchanova) [reply python/pyspark/sql/window.py]: As I can see orderBy is implemented in scala.
It accepts sorting order via string query, something like 'col_name desc'

Comment (HyukjinKwon) [reply python/pyspark/sql/window.py]: Would you mind if I ask to point out the corresponding API in Scala side?

Comment (annamolchanova) [reply python/pyspark/sql/window.py]: It's in the expressions pachage:
http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.sql.expressions.Window$

Comment (HyukjinKwon) [reply python/pyspark/sql/window.py]: `orderBy` looks not taking `ascending` in the API documentation.
screen shot 2018-09-25 at 5 33 51 pm

Comment (HyukjinKwon) [reply python/pyspark/sql/window.py]: We match it to the equivalent API. If it's not in Scale side; we should add it in Scale side first.

File: python/pyspark/sql/column.py [hunk 3]
```diff
@@ -54,6 +54,22 @@ def _to_java_column(col):
...
+def _to_sorted_java_columns(cols, ascending=True):
+    if len(cols) == 1 and isinstance(cols[0], list):
+        cols = cols[0]
+    jcols = [_to_java_column(c) for c in cols]
+    if isinstance(ascending, (bool, int)):
+        if not ascending:
+            jcols = [jc.desc() for jc in jcols]
+    elif isinstance(ascending, list):
+        jcols = [jc if asc else jc.desc()
+                 for asc, jc in zip(ascending, jcols)]
```
Comment (HyukjinKwon) [inline python/pyspark/sql/column.py]: Btw this can be easily worked around. I think we don't need to add this argument

**Comment by holdenk** [comment]
Jenkins OK to test.
@annamolchanova if you want help making the Scala version of this PR first I'd be happy to lend what help I can.

**Comment by HyukjinKwon** [comment]
I don't think we should add this .. this can be easily worked around .. The PR is not ready and the author looks inactive anyway. Let me leave this closed.

**Pull request closed without merging**